src/Compiler/Reports/PlacementReportManager.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2022 The Foedag team
+
+GPL License
+
+Copyright (c) 2022 The Open-Source FPGA Foundation
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "PlacementReportManager.h"
+
+#include <QFile>
+#include <QRegularExpression>
+#include <QTextStream>
+
+#include "CompilerDefines.h"
+#include "NewProject/ProjectManager/project.h"
+#include "Utils/FileUtils.h"
+
+namespace {
+static constexpr const char *REPORT_NAME{"Report Resource Utilization"};
+static constexpr const char *RESOURCES_SPLIT{"blocks of type:"};
+
+static constexpr const char *BLOCKS_COL{"Blocks"};
+static constexpr const char *NETLIST_COL{"Netlist"};
+static constexpr const char *ARCHITECTURE_COL{"Architecture"};
+
+static const QRegExp FIND_TIMINGS{
```

Comment:
When we revisit the reports, we should consider switching to QRegularExpression as QRegExp class is deprecated in Qt6. Additionally, QRegularExpression has better support for perl regex so the syntax issues you were bumping into might be resolved by using QRegularExpression instead.

Reply:
Why can't we use now? Wondering if it's a lot of refactoring. We got other reports to build on.

Reply:
It sounded like a number of these parsers are going to get some level of rewrite once the reporting format is more determined so i figured it'd be easier to make the changes then instead of introducing new logic mid PR as that would require re-testing etc.
Not sure how much of a change it is, [this section ](https://doc.qt.io/qt-5/qregularexpression.html#notes-for-qregexp-users) makes it look like there are a couple changes to consider.

Reply:
Thanks.